services/rollout-service/pkg/service/broadcast.go

```diff
@@ -192,29 +196,85 @@ func (b *Broadcast) StreamStatus(req *api.StreamStatusRequest, svc api.RolloutSe
 }
 
 func (b *Broadcast) GetStatus(ctx context.Context, req *api.GetStatusRequest) (*api.GetStatusResponse, error) {
-	resp, _, unsubscribe := b.Start()
+	var wait <-chan time.Time
+	if req.WaitSeconds > 0 {
+		wait = time.After(time.Duration(req.WaitSeconds) * time.Second)
+	}
+	resp, ch, unsubscribe := b.Start()
 	defer unsubscribe()
-	apps := []*api.GetStatusResponse_ApplicationStatus{}
-	status := api.RolloutStatus_RolloutStatusSuccesful
+	apps := map[Key]*api.GetStatusResponse_ApplicationStatus{}
 	for _, r := range resp {
-		if r.EnvironmentGroup == req.EnvironmentGroup {
-			s := getStatus(r)
-			if s.RolloutStatus == api.RolloutStatus_RolloutStatusSuccesful {
-				continue
+		s := filterApplication(req, r)
+		if s != nil {
+			apps[r.Key] = s
+		}
+	}
+	status := aggregateStatus(apps)
+	if wait != nil {
+		// The waiting function is used in testing to make sure, we are really processing delayed events.
```

Comment:
The comment makes it sound a bit like waiting is only used in testing? Is that accurate? Maybe rephrase.

Reply:
Yes, that is correct.